tests/python_client/testcases/test_index.py

```diff
@@ -1547,6 +1548,261 @@ def test_index_with_string_field_empty(self):
+        collection_w = self.init_collection_wrap(name=c_name)
+        data = cf.gen_default_list_data()
+        collection_w.insert(data=data)
+        index, _ = self.index_wrap.init_index(collection_w.collection, default_float_vec_field_name, ct.default_diskann_index)
+        log.info(self.index_wrap.params)
+        cf.assert_equal_index(index, collection_w.indexes[0])
+        collection_w.load()
+        assert collection_w.num_entities == default_nb
+        vectors = [[random.random() for _ in range(default_dim)] for _ in range(default_nq)]
+        search_res, _=collection_w.search(vectors[:default_nq], default_search_field,
+                            ct.default_diskann_search_params, default_limit,
+                            default_search_exp)
+        assert len(search_res) == default_nq
+        assert len(search_res[0]) == default_limit
+
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_create_index_with_over_max_dim(self):
+        """
+        target: test create index with diskann
+        method: 1.create collection, when the dim of the vector is greater than 1024 
+                2.create diskann index
+        expected: create index raise an error
+        """
+        dim = 1025
+        c_name = cf.gen_unique_str(prefix)
+        fields = [cf.gen_int64_field(is_primary=True), cf.gen_float_field(), cf.gen_string_field(),
+                  cf.gen_float_vec_field(dim=dim)]
+        schema = cf.gen_collection_schema(fields=fields)
+        collection_w = self.init_collection_wrap(name=c_name, schema=schema)
+        collection_w.create_index(default_float_vec_field_name, ct.default_diskann_index,
+                                  check_task=CheckTasks.err_res,
+                                  check_items={ct.err_code: 1,
+                                               ct.err_msg: "invalid index params"})
+
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_create_index_with_diskann_callback(self,_async):
+        """
+        target: test create index with diskann
+        method: 1.create collection and insert data 
+                2.create diskann index ,then load
+                3.search 
+        expected: create index successfully
+        """
+        c_name = cf.gen_unique_str(prefix)
+        collection_w = self.init_collection_wrap(c_name)
+        data = cf.gen_default_list_data()
+        collection_w.insert(data=data)
+        res, _ = collection_w.create_index(ct.default_float_vec_field_name, ct.default_diskann_index,
+                                           index_name=ct.default_index_name, _async=_async,
+                                           _callback=self.call_back())
+
+        if _async:
+            res.done()
+            assert len(collection_w.indexes) == 1
+        collection_w.load()
+        vectors = [[random.random() for _ in range(default_dim)] for _ in range(default_nq)]
+        search_res, _=collection_w.search(vectors[:default_nq], default_search_field,
```

Comment:
Here should format this line.

And why not use search result check function directly?

Reply:
Formatting has been changed